src/goto-instrument/cover.cpp

```diff
@@ -285,16 +291,22 @@ static void instrument_cover_goals(
       // Simplify the common case where we have ASSERT(x); ASSUME(x):
       if(i_it->is_assert())
       {
-        auto successor = std::next(i_it);
-        if(
-          successor != function.body.instructions.end() &&
-          successor->is_assume() &&
-          successor->get_condition() == i_it->get_condition())
+        if(!cover_config.cover_failed_assertions)
         {
-          successor->turn_into_skip();
+          auto successor = std::next(i_it);
+          if(
+            successor != function.body.instructions.end() &&
+            successor->is_assume() &&
+            successor->get_condition() == i_it->get_condition())
+          {
+            successor->turn_into_skip();
+          }
+          i_it->type = goto_program_instruction_typet::ASSUME;
+        }
+        else
+        {
+          i_it->turn_into_skip();
```

Comment:
Btw we have to skip them rather than just leaving them be because otherwise cbmc would consider these assertions coverage goals.

Reply:
Why not make `--no-assertions` (aka turn-assertions-into-skips) useable with `--cover` then (as @martin-cs  suggests)? That would make the behaviour more transparent to the user.